src/lattice_reaction_system_diffusion.jl

```diff
@@ -0,0 +1,507 @@
+               (comp - 1) * lrs.nS
+        if in(s, diff_species)
+            # Finds the location of the diffusion elements, and inserts the elements from the non-spatial part into this.
+            diffusion_rows = (lrs.lattice.fadjlist[comp] .- 1) .* lrs.nS .+ s
+            split_idx = isempty(rows) ? 1 : findfirst(diffusion_rows .> rows[1])
+            isnothing(split_idx) && (split_idx = length(diffusion_rows) + 1)
+            rows = vcat(diffusion_rows[1:(split_idx - 1)], rows,
+                        diffusion_rows[split_idx:end])
+            if only_diff[s]
+                split_idx = findfirst(rows .> get_index(comp, s, lrs.nS))
+                isnothing(split_idx) && (split_idx = length(rows) + 1)
+                insert!(rows, split_idx, get_index(comp, s, lrs.nS))
+            end
+        end
+        J_rowval[J_colptr[col_idx]:(J_colptr[col_idx + 1] - 1)] = rows
+    end
+
+    # Set element values.
+    if !set_nonzero
+        J_nzval .= 1.0
+    else
+        for (s_idx, (s, rates)) in enumerate(diffusion_rates),
+            (e_idx, edge) in enumerate(edges(lrs.lattice))
+
+            col_start = J_colptr[get_index(edge.src, s, lrs.nS)]
+            col_end = J_colptr[get_index(edge.src, s, lrs.nS) + 1] - 1
+            column_view = @view J_rowval[col_start:col_end]
+
+            # Updates the source value.
+            val_idx_src = col_start +
+                          findfirst(column_view .== get_index(edge.src, s, lrs.nS)) - 1
+            J_nzval[val_idx_src] -= get_component_value(rates, e_idx)
+
+            # Updates the destination value.
+            val_idx_dst = col_start +
+                          findfirst(column_view .== get_index(edge.dst, s, lrs.nS)) - 1
+            J_nzval[val_idx_dst] += get_component_value(rates, e_idx)
+        end
+    end
+
+    return SparseMatrixCSC(lrs.nS * lrs.nC, lrs.nS * lrs.nC, J_colptr, J_rowval, J_nzval)
+end
+
+### JumpProblem ###
+
+# Builds a spatial DiscreteProblem from a Lattice Reaction System.
+
+# Creates a DiscreteProblem from a LatticeReactionSystem.
+function DiffEqBase.DiscreteProblem(lrs::LatticeReactionSystem, u0_in, tspan,
+                                    p_in = DiffEqBase.NullParameters(), args...; kwargs...)
+    u0 = lattice_process_u0(u0_in, ModelingToolkit.getname.(species(lrs)), lrs.nC)
+    pC, pD = lattice_process_p(p_in, Symbol.(compartment_parameters(lrs)),
+                               Symbol.(diffusion_parameters(lrs)), lrs)
+    return DiscreteProblem(lrs.rs, u0, tspan, (pC, pD), args...; kwargs...)
+end
+
+# Builds a spatial JumpProblem from a DiscreteProblem containg a Lattice Reaction System.
+function JumpProcesses.JumpProblem(lrs::LatticeReactionSystem, dprob, aggregator, args...;
+                                   name = nameof(lrs.rs),
+                                   combinatoric_ratelaws = get_combinatoric_ratelaws(lrs.rs),
+                                   checks = false, kwargs...)
```

Comment:
Instead of asking the user a question, let's help them understand what they did wrong and how to fix it.
```suggestion
        error("Parameters in input DiscreteProblem is of an unexpected type: $(typeof(dprob.p)). This might be caused by passing a `LatticeReactionProblem` into the `DiscreteProblem`. Pass a `ReactionProblem` instead.")
```